subprojects/launcher/src/main/java/org/gradle/launcher/cli/DefaultCommandLineActionFactory.java

```diff
@@ -367,6 +367,9 @@ public void execute(ExecutionListener executionListener) {
                 loggingManager.attachProcessConsole(loggingConfiguration.getConsoleOutput());
                 new WelcomeMessageAction(buildLayout).execute(Logging.getLogger(WelcomeMessageAction.class));
                 exceptionReportingAction.execute(executionListener);
+            } catch (RuntimeException e) {
+                reporter.execute(e);
+                executionListener.onFailure(e);
```

Comment:
This catch block uses the reporter that has logging configured using the start parameter values.
Without it, the exception is caught on a higher level where a new reporter is initialized using default values making `--stacktrace` have no effect. This can be observed when native services fail to initialize.

Reply:
Could we move
```
                    NativeServices.initialize(buildLayout.getGradleUserHomeDir());
                    loggingManager.attachProcessConsole(loggingConfiguration.getConsoleOutput());          
                    new WelcomeMessageAction(buildLayout).execute(Logging.getLogger(WelcomeMessageAction.class));
```
into a new `Action<ExecutionListener>` (NativeServicesInitializeAction) that delegates to `action` and have `exceptionReportingAction` delegate to that?  That would make ExceptionReportingAction do the right thing.

The order would be:

ExceptionReportingAction.execute calls NativeServicesInitializeAction.execute calls action.execute